src/classes/CASE_CannotDelete_TEST.cls

```diff
@@ -0,0 +1,109 @@
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Cases
+* @group-content ../../ApexDocContent/Case.htm
+* @description Unit tests for CASE_CannotDelete_TDTM. These tests
+* make sure Cases cannot be deleted when they have Behavior Involvement Associations.
+*/
+
+@isTest
+private class CASE_CannotDelete_TEST {
+
+    @testSetup
+    static void dataSetup() {
+        
+        // Insert Contacts
+        List<Contact> students = new List<Contact>();
+        
+        Contact student1 = UTIL_UnitTestData_TEST.getContact();
+        Contact student2 = UTIL_UnitTestData_TEST.getContact();
+        
+        students.add(student1);
+        students.add(student2);
+
+        insert students;
+
+        // Insert Cases
+        List<Case> cases = new List<Case>();
+        Case case1 = UTIL_UnitTestData_TEST.getCase(student1.Id, 'School Violation', 'New');
+        Case case2 = UTIL_UnitTestData_TEST.getCase(student2.Id, 'School Violation', 'New');
+
+        cases.add(case1);
+        cases.add(case2);
+
+        insert cases;
+
+        // Insert Behavior Involvement records
+        List<Behavior_Involvement__c> behaviorRecords = new List<Behavior_Involvement__c>();
+
+        Behavior_Involvement__c behavInvolRec1 = UTIL_UnitTestData_TEST.getBehaviorInvolvementRecord(student1.Id, case1.Id, 'Victim');
+        Behavior_Involvement__c behavInvolRec2 = UTIL_UnitTestData_TEST.getBehaviorInvolvementRecord(student2.Id, case2.Id, 'Witness');
+
+        behaviorRecords.add(behavInvolRec1);
+        behaviorRecords.add(behavInvolRec2);
+
+        insert behaviorRecords;
+    }
+
+
+    /*************************************************************************************************************
+    * @description Test deletion of Cases with Behavior Involvement and preventing deletion setting is turned On
+    */
+    @isTest
+    static void testCaseDeleteSettingOn() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Prevent_Case_Deletion__c = TRUE));
+
+        Test.startTest();
+            List<Case> casesToDelete = [SELECT Id FROM Case];
+            Database.DeleteResult[] results = Database.delete(casesToDelete, false);
+        Test.stopTest();
+
+        // Ensure no Cases are deleted
+        List<Case> casesAfterDelete = [SELECT Id FROM Case];
+        System.assertEquals(2, casesAfterDelete.size());
+    }
+
+    /*************************************************************************************************************
+    * @description Test deletion of Cases with Behavior Involvement and preventing deletion setting is turned Off
+    */
+    @isTest
+    static void testCaseDeleteSettingOff() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Prevent_Case_Deletion__c = False));
```

Comment:
NIT: very minor, we should probably as a team figure out if we want to do true|false, True|False, or TRUE|FALSE, but we probably should be consistent.

Reply:
@lifewithryan Thank you. I fixed it, can you give it another shot?